client/src/components/FieldAgents.js

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from "react";
+import AddFieldAgent from "./AddFieldAgent";
+import DeleteFieldAgent from "./DeleteFieldAgent";
+import UpdateFieldAgent from "./UpdateFieldAgent";
+
+
+const VIEW = {
+  LIST: 'list',
+  ADD: 'add',
+  EDIT: 'edit',
+  DELETE: 'delete'
+};
+
+function FieldAgents() {
+
+  const [fieldAgents, setFieldAgents] = useState([]);
+  const [fieldAgent, setFieldAgent] = useState({});
+  const [view, setView] = useState(VIEW.LIST);
+
+  const getFieldAgent = () => {
+    fetch('http://localhost:8080/api/agent')
+      .then(response => {
+        if (response.status === 200) {
+          return response.json()
+        }
+
+        return Promise.reject('Something went wrong on the server :)');
+      })
+      .then(body => {
+        setFieldAgents(body);
+      })
+      .catch(err => console.error(err));
+  }
+
+  useEffect(() =>{
+    getFieldAgent();
+  }, []);
+
+
+  const handleAdd = (newFieldAgent) => {
+    setFieldAgents([...fieldAgents, newFieldAgent]);
```

Comment:
On add, update, and delete, it's also okay to re-fetch all agents with `getFieldAgents`.
When a lot of users are editing data, it makes sense to periodically "refresh" our data.